runtime/compiler/optimizer/JProfilingValue.cpp

```diff
@@ -193,104 +193,129 @@ TR_JProfilingValue::perform()
+         }
+      }
+   // For Virtual call and class test nodes, we are actually profiling the VFT pointer of the object. So We need to set the Bit VE
+   if (preceedingTT != NULL)
+      {
+      alreadyProfiledValues->set(node->getFirstChild()->getGlobalIndex());
+      TR::Node *call = TR::Node::createWithSymRef(node, TR::call, 2, profiler);
+      call->setAndIncChild(0, profiledNode);
+      TR_ValueProfileInfo *valueProfileInfo = TR_PersistentProfileInfo::getCurrent(comp())->findOrCreateValueProfileInfo(comp());
+      TR_AbstractHashTableProfilerInfo *info = static_cast<TR_AbstractHashTableProfilerInfo*>(valueProfileInfo->getOrCreateProfilerInfo(profiledNode->getByteCodeInfo(), comp(), AddressInfo, HashTableProfiler));
+      call->setAndIncChild(1, TR::Node::aconst(node, (uintptr_t) info));
+      TR::TreeTop *callTree = TR::TreeTop::create(comp(), preceedingTT, TR::Node::create(TR::treetop, 1, call));
+      callTree->getNode()->setIsProfilingCode();
+      }
+   
    for (int i = 0; i < node->getNumChildren(); ++i)
-      performOnNode(node->getChild(i), tt, checklist);
-   }
-
-void
-TR_JProfilingValue::addProfiling(TR::Node *value, TR::TreeTop *tt)
-   {
-   if (!performTransformation(comp(), "%s Add trees to track the value of node %p near tree %p\n", optDetailString(), value, tt->getNode()))
-      return;
-
-   TR_ValueProfileInfo *valueProfileInfo = TR_PersistentProfileInfo::getCurrent(comp())->findOrCreateValueProfileInfo(comp());
-   TR_AbstractHashTableProfilerInfo *info = static_cast<TR_AbstractHashTableProfilerInfo*>(valueProfileInfo->getOrCreateProfilerInfo(value->getByteCodeInfo(), comp(), AddressInfo, HashTableProfiler));
-   addProfilingTrees(comp(), tt, value, info, NULL, true, trace());
+      performOnNode(node->getChild(i), cursor, alreadyProfiledValues, checklist);
    }
 
-void
-TR_JProfilingValue::addVFTProfiling(TR::Node *address, TR::TreeTop *tt, bool addNullCheck)
-   {
-   if (!performTransformation(comp(), "%s Add trees to track the vft lookup of node %p near tree %p, null check %d\n", optDetailString(), address, tt->getNode(), addNullCheck))
-      return;
-
-   TR::Node *vftNode = TR::Node::createWithSymRef(address, TR::aloadi, 1, address,
-      getSymRefTab()->findOrCreateVftSymbolRef());
-
-   TR::Node *check = NULL;
-   if (addNullCheck)
-      check = TR::Node::createif(TR::ifacmpeq, address, TR::Node::aconst(address, 0));
-
-   TR_ValueProfileInfo *valueProfileInfo = TR_PersistentProfileInfo::getCurrent(comp())->findOrCreateValueProfileInfo(comp());
-   TR_AbstractHashTableProfilerInfo *info = static_cast<TR_AbstractHashTableProfilerInfo*>(valueProfileInfo->getOrCreateProfilerInfo(address->getByteCodeInfo(), comp(), AddressInfo, HashTableProfiler));
-   addProfilingTrees(comp(), tt, vftNode, info, check, true, trace());
-   }
 
 /**
  * Identify helper calls to TR_jProfile32BitValue and TR_jProfile64BitValue, lowering them
```

Comment:
This comment needs an update I think since the symrefs have changed? You have also removed the comment markers showing the sections of the code for the mainline and various helper blocks - could add something like that back in to make it easier to know what each section of code is building?

Reply:
Moved brief to Header file and changed comment : https://github.com/eclipse/openj9/pull/6585/commits/4babc9149b99cc98d9703d5340c8c21f81c918da